src/plugins/internalnotification/internalnotification.c

```diff
@@ -0,0 +1,263 @@
+/**
+ * @file
+ *
+ * @brief Source for internalnotification plugin
+ *
+ * @copyright BSD License (see doc/COPYING or http://www.libelektra.org)
+ *
+ */
+
+#include "internalnotification.h"
+
+#if PLUGIN_INTERNALNOTIFICATION_VERBOSE
+#include <stdio.h>
+#endif
+
+#include <errno.h>
+#include <stdlib.h>
+
+#include <kdbhelper.h>
+
+struct _KeyRegistration
+{
+	char * name;
+	int * variable;
+	struct _KeyRegistration * next;
+};
+typedef struct _KeyRegistration KeyRegistration;
+
+struct _ListPointer
+{
+	KeyRegistration * head;
+	KeyRegistration * tail;
```

Comment:
A better name for `tail` might be `last`. The tail of a list is usually everything except for the first element:

![listmonster](https://cloud.githubusercontent.com/assets/691989/17204483/169cd90e-54a6-11e6-917b-7b1834b4dc2d.png)

Image Source: http://learnyouahaskell.com/starting-out#an-intro-to-lists